From ba34d29274c23c52be957ea040539dccbab09765 Mon Sep 17 00:00:00 2001
From: Alper Nebi Yasak <alpernebiyasak@gmail.com>
Date: Wed, 10 Jul 2024 17:37:56 +0300
Subject: [PATCH] HACK: Makefile: Ignore missing input files for binman images

For Rockchip boards binman tries to build SPI and MMC images that
require an externally built BL31 file to be provided, and the build
fails otherwise.

Some downstreams only care about build outputs for U-Boot proper. As a
hack to make sure they can do so without passing in a BL31 file, tell
binman to ignore missing input files.

Signed-off-by: Alper Nebi Yasak <alpernebiyasak@gmail.com>
---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 1c754ceb5953..fec8d9b2d2b1 100644
--- a/Makefile
+++ b/Makefile
@@ -1375,7 +1375,7 @@ cmd_binman = $(srctree)/tools/binman/binman $(if $(BINMAN_DEBUG),-D) \
                 --toolpath $(objtree)/tools \
 		$(if $(BINMAN_VERBOSE),-v$(BINMAN_VERBOSE)) \
 		build -u -d u-boot.dtb -O . -m \
-		--allow-missing $(if $(BINMAN_ALLOW_MISSING),--ignore-missing) \
+		--allow-missing --ignore-missing \
 		-I . -I $(srctree) -I $(srctree)/board/$(BOARDDIR) \
 		-I $(dt_dir) -a of-list=$(CONFIG_OF_LIST) \
 		$(foreach f,$(BINMAN_INDIRS),-I $(f)) \
-- 
2.45.2

